voila/handler.py

```diff
@@ -198,12 +198,18 @@ def time_out():
 
                 return "<script>window.voila_heartbeat()</script>\n"
 
+            server_url = self.request.full_url()
+            # Removing trailing slash
+            if server_url[-1] == "/":
+                server_url = server_url[:-1]
+
             kernel_env = {**os.environ, **request_info}
             kernel_env[ENV_VARIABLE.VOILA_PREHEAT] = "False"
             kernel_env[ENV_VARIABLE.VOILA_BASE_URL] = self.base_url
-            kernel_env[ENV_VARIABLE.VOILA_SERVER_URL] = self.settings.get(
-                "server_url", "/"
-            )
+            kernel_env[ENV_VARIABLE.VOILA_SERVER_URL] = server_url
```

Comment:
We also set these variables in the preheated kernels, should we update it to make it consistent with this case?

Reply:
I mean the value of the variables should be the same in both cases, we can keep your implementation in `voila/handler.py` and then update the corresponding variable in `VoilaKernelManager`